hmpps-sqs-spring-boot-autoconfigure/src/test/kotlin/uk/gov/justice/hmpps/sqs/HmppsAuditServiceTest.kt

```diff
@@ -32,23 +34,156 @@ internal class HmppsAuditServiceTest(@Autowired private val objectMapper: Object
     whenever(hmppsQueueService.findByQueueId(any())).thenReturn(hmppsQueue)
     whenever(hmppsQueue.sqsClient).thenReturn(sqsAsyncClient)
     whenever(hmppsQueue.queueUrl).thenReturn("a queue url")
-  }
 
-  @Test
-  internal fun testPublishEvent() = runTest {
     whenever(sqsAsyncClient.sendMessage(any<SendMessageRequest>()))
       .thenReturn(CompletableFuture.completedFuture(SendMessageResponse.builder().build()))
+  }
+
+  @Nested
+  inner class PublishEventUsingDomainObject {
+
+    @Test
+    internal fun `test publish event using domain object`() = runTest {
```

Comment:
A side effect of using the internal keyword for tests in nested classes is that if you click on the test name in the Intellij Run window it doesn't take you to the test in the editor window. Hence I never use internal and don't know why Intellij suggests it in the first place.

Reply:
interesting, didn't realise that.  I've seen that in the past and wondered what was going wrong